backend/frontend/tests/test_restricted_user_board_access.py

```diff
@@ -0,0 +1,159 @@
+        )
+        self.user_with_manage_card_permission.set_password(
+            self.user_with_manage_card_permission.email
+        )
+        self.user_with_manage_card_permission.save()
+        assign_perm(
+            Team.PERMISSION_MANAGE_CARDS,
+            self.user_with_manage_card_permission,
+            self.viewed_user_team,
+        )
+
+    def _setup_user_with_view_permission(self):
+        self.user_with_view_permission = UserFactory(
+            email="user_with_view_permission.umuzi.org"
+        )
+        self.user_with_view_permission.set_password(
+            self.user_with_view_permission.email
+        )
+        self.user_with_view_permission.save()
+        assign_perm(
+            Team.PERMISSION_VIEW_ALL,
+            self.user_with_view_permission,
+            self.viewed_user_team,
+        )
+
+    def _setup_user_with_review_cards_permission(self):
+        self.user_with_review_cards_permission = UserFactory(
+            email="user_with_review_cards_permission.umuzi.org"
+        )
+        self.user_with_review_cards_permission.set_password(
+            self.user_with_review_cards_permission.email
+        )
+        self.user_with_review_cards_permission.save()
+        assign_perm(
+            Team.PERMISSION_REVIEW_CARDS,
+            self.user_with_review_cards_permission,
+            self.viewed_user_team,
+        )
+
+    def _setup_trusted_user(self):
+        self.trusted_reviewer = UserFactory(email="trusted_reviewer.umuzi.org")
+        self.trusted_reviewer.set_password(self.trusted_reviewer.email)
+        self.trusted_reviewer.save()
+        assign_perm(
+            Team.PERMISSION_TRUSTED_REVIEWER,
+            self.trusted_reviewer,
+            self.viewed_user_team,
+        )
+
+    def test_super_user_can_view_user_board(self):
+        self.do_login(self.superuser)
+        url = self.reverse_url("user_board", kwargs={"user_id": self.viewed_user.id})
+        self.page.goto(url)
+
+        body = self.page.text_content("body")
+
+        self.assertIn(f"Viewing {self.viewed_user.email}", body)
+
+    def test_user_can_view_own_board(self):
+        self.do_login(self.viewed_user)
+        url = self.reverse_url("user_board", kwargs={"user_id": self.viewed_user.id})
+        self.page.goto(url)
+
+        body = self.page.text_content("body")
+
+        self.assertIn(f"Viewing {self.viewed_user.email}", body)
+
+    def test_user_without_view_access_cannot_view_other_user_board(self):
+        """
+        @user_passes_test(func) mixin we use to control access
+        redirects user to login page if they don't pass the test.
+        """
+        self.do_login(self.user_without_access)
+        url = self.reverse_url("user_board", kwargs={"user_id": self.viewed_user.id})
+
+        self.page.goto(url)
+
+        login_url = "/login"
+        final_page_url = self.page.url
+
+        self.assertIn(login_url, final_page_url)
```

Comment:
What if the user is already logged in? Should something else happen? Eg, maybe they should get a page like "you do not have permission to view this page"